swarm/swarm.go

```diff
@@ -201,8 +201,9 @@ func NewSwarm(config *api.Config, mockStore *mock.NodeStore) (self *Swarm, err e
 	}
 	resourceHandler.SetStore(netStore)
 
-	var validators []storage.ChunkValidator
-	validators = append(validators, storage.NewContentAddressValidator(storage.MakeHashFunc(storage.DefaultHash)))
+	validators := []storage.ChunkValidator{
```

Comment:
@nolash @gbalint we should rename this, as this is confusing.

What we are doing here is not chunk validation, but rather chunk classification, and then validating chunks based on their `type`:
1. If it is a `content addressed chunk` -> validate the chunk according to the rules for content addressed chunks
2. If it is a `resource chunk` -> validate according to the resource chunk rules

The slice of `ChunkValidator` suggests that this list of Validators will be applied to the chunk, which is far from true :)

Probably a good idea to introduce `types` of chunks, where each `type` has a set of validators that are applied to it, in order to determine if the chunk is valid or not.